### DISCUSSION TITLE
IN_OPENRESULT_NOMEMORY or IN_OPENRESULT_UNKNOWNERROR in DryWetMidi

### DISCUSSION THREAD

**Discussion by mystery123sk** [melanchall/drywetmidi#288] [General]
Hi,

I receive quite often these kind of crashes in DryWetMidi from my clients' computers:

Example 1.
2/4/2024 11:00:06 PM I: Midi closed IN: SoundSwitch Control One (id: 436, started: False)
2/4/2024 11:00:06 PM I: Midi closed OUT: Microsoft GS Wavetable Synth (id: 437, started: False)
2/4/2024 11:00:06 PM I: Midi closed OUT: SoundSwitch Control One (id: 438, started: False)
2/4/2024 11:00:06 PM E: OnDispatcherUnhandledExceptionEventHandler on thread ID 1
2/4/2024 11:00:06 PM E: System.Windows.Threading.DispatcherUnhandledExceptionEventArgs
2/4/2024 11:00:06 PM E: Inner exception:
2/4/2024 11:00:06 PM E: Melanchall.DryWetMidi.Multimedia.MidiDeviceException: There is no memory in the system to complete the operation (IN_OPENRESULT_NOMEMORY).
v Melanchall.DryWetMidi.Multimedia.NativeApi.HandleResult[TResult,TException](TResult result, Func`3 createException)
v Melanchall.DryWetMidi.Multimedia.InputDevice.EnsureHandleIsCreated()
v Melanchall.DryWetMidi.Multimedia.InputDevice.StartEventsListening()

Example 2
11/29/2023 4:27:16 PM I: Midi closed IN: APC mini mk2 (id: 17, started: False)
11/29/2023 4:27:16 PM I: Midi closed IN: MIDIIN2 (APC mini mk2) (id: 18, started: False)
11/29/2023 4:27:16 PM I: Midi closed OUT: Microsoft GS Wavetable Synth (id: 19, started: False)
11/29/2023 4:27:16 PM I: Midi closed OUT: APC mini mk2 (id: 20, started: False)
11/29/2023 4:27:16 PM I: Midi closed OUT: MIDIOUT2 (APC mini mk2) (id: 21, started: False)
11/29/2023 4:27:16 PM E: OnDispatcherUnhandledExceptionEventHandler on thread ID 1
11/29/2023 4:27:16 PM E: System.Windows.Threading.DispatcherUnhandledExceptionEventArgs
11/29/2023 4:27:16 PM E: Inner exception:
11/29/2023 4:27:16 PM E: Melanchall.DryWetMidi.Multimedia.MidiDeviceException: Internal error (IN_OPENRESULT_UNKNOWNERROR).
at Melanchall.DryWetMidi.Multimedia.NativeApi.HandleResult[TResult,TException](TResult result, Func`3 createException)
at Melanchall.DryWetMidi.Multimedia.InputDevice.EnsureHandleIsCreated()
at Melanchall.DryWetMidi.Multimedia.InputDevice.StartEventsListening()

Example 3
11/10/2023 7:12:17 PM I: Midi closed OUT: Microsoft GS Wavetable Synth (id: 3, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: Microsoft GS Wavetable Synth (id: 4, started: False)
11/10/2023 7:12:17 PM I: Midi opened IN: MIDIIN2 (LPMiniMK3 MIDI) (id: 6, started: False)
11/10/2023 7:12:17 PM I: Midi closed IN: LPMiniMK3 MIDI (id: 5, started: False)
11/10/2023 7:12:17 PM I: Midi closed IN: MIDIIN2 (LPMiniMK3 MIDI) (id: 6, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: Microsoft GS Wavetable Synth (id: 7, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: LPMiniMK3 MIDI (id: 8, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: MIDIOUT2 (LPMiniMK3 MIDI) (id: 9, started: False)
11/10/2023 7:12:17 PM I: Midi opened IN: MIDIIN2 (LPMiniMK3 MIDI) (id: 11, started: False)
11/10/2023 7:12:17 PM I: Midi closed IN: LPMiniMK3 MIDI (id: 10, started: False)
11/10/2023 7:12:17 PM I: Midi closed IN: MIDIIN2 (LPMiniMK3 MIDI) (id: 11, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: Microsoft GS Wavetable Synth (id: 12, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: LPMiniMK3 MIDI (id: 13, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: MIDIOUT2 (LPMiniMK3 MIDI) (id: 14, started: False)
11/10/2023 7:12:17 PM I: Midi opened IN: MIDIIN2 (LPMiniMK3 MIDI) (id: 16, started: False)
11/10/2023 7:12:17 PM I: Midi closed IN: LPMiniMK3 MIDI (id: 15, started: False)
11/10/2023 7:12:17 PM I: Midi closed IN: MIDIIN2 (LPMiniMK3 MIDI) (id: 16, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: Microsoft GS Wavetable Synth (id: 17, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: LPMiniMK3 MIDI (id: 18, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: MIDIOUT2 (LPMiniMK3 MIDI) (id: 19, started: False)
11/10/2023 7:12:17 PM I: Midi opened IN: MIDIIN2 (LPMiniMK3 MIDI) (id: 21, started: False)
11/10/2023 7:12:17 PM I: Midi closed IN: LPMiniMK3 MIDI (id: 20, started: False)
11/10/2023 7:12:17 PM I: Midi closed IN: MIDIIN2 (LPMiniMK3 MIDI) (id: 21, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: Microsoft GS Wavetable Synth (id: 22, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: LPMiniMK3 MIDI (id: 23, started: False)
11/10/2023 7:12:17 PM I: Midi closed OUT: MIDIOUT2 (LPMiniMK3 MIDI) (id: 24, started: False)
11/10/2023 7:12:17 PM I: Midi opened IN: MIDIIN2 (LPMiniMK3 MIDI) (id: 26, started: False)
11/10/2023 7:13:38 PM I: Midi closed IN: LPMiniMK3 MIDI (id: 25, started: False)
11/10/2023 7:13:38 PM I: Midi closed IN: MIDIIN2 (LPMiniMK3 MIDI) (id: 26, started: False)
11/10/2023 7:13:38 PM I: Midi closed OUT: Microsoft GS Wavetable Synth (id: 27, started: False)
11/10/2023 7:13:38 PM I: Midi closed OUT: LPMiniMK3 MIDI (id: 28, started: False)
11/10/2023 7:13:38 PM I: Midi closed OUT: MIDIOUT2 (LPMiniMK3 MIDI) (id: 29, started: False)
11/10/2023 7:13:38 PM E: OnDispatcherUnhandledExceptionEventHandler on thread ID 1
11/10/2023 7:13:38 PM E: System.Windows.Threading.DispatcherUnhandledExceptionEventArgs
11/10/2023 7:13:38 PM E: Inner exception:
11/10/2023 7:13:38 PM E: Melanchall.DryWetMidi.Multimedia.MidiDeviceException: Internal error (IN_OPENRESULT_UNKNOWNERROR).
a következő helyen: Melanchall.DryWetMidi.Multimedia.NativeApi.HandleResult[TResult,TException](TResult result, Func`3 createException)
a következő helyen: Melanchall.DryWetMidi.Multimedia.InputDevice.EnsureHandleIsCreated()
a következő helyen: Melanchall.DryWetMidi.Multimedia.InputDevice.StartEventsListening()

Unfortunately I can't reproduce the bugs, since I receive them as crashlogs from clients' computers. Any idea if I'm doing something wrong at initialization?

Michal

**Comment by melanchall** [comment]
Hi,

> Any idea if I'm doing something wrong at initialization?

But.. What are you doing? :-) You didn't provide any code. Especially how you open and close devices.

But these errors are not bugs within the DryWetMIDI. They come from Windows MIDI system. DryWetMIDI just redirects them to you.
